src/contexts/UserContext.jsx

```diff
@@ -0,0 +1,25 @@
+import { createContext, useState, useEffect } from "react";
+import { getUsers } from "../../utils/api";
+
+export const UserContext = createContext();
+
+export const UserProvider = (props) => {
+  const [users, setUsers] = useState([]);
+  const [currUser, setCurrUser] = useState(() => {
+    const saved = localStorage.getItem("user");
```

Comment:
Pretty cool use of localstorage, user persists despite a refresh which is nice :-)